### PR TITLE
Initial commit of actor load test app

### PR DESCRIPTION
**Pull request by youngbupark**
# Description

This is the initial commit of actor load test application and driver:

* stateactor actor service application: implements `setActorState` and `getActorState` actor method to save and get the state.
* testclient application: uses [fortio](https://github.com/fortio/fortio) to generate load and create the report.

## Next steps

* Run test

## Test examples

```
% ./testclient --help
Usage of ./testclient:
  -a string
        Actor Type (default "StateActor")
  -c int
        Number of parallel simultaneous connections. (default 10)
  -logcaller
        Logs filename and line number of callers to log (default true)
  -loglevel value
        loglevel, one of [Debug Verbose Info Warning Error Critical Fatal] (default Info)
  -logprefix string
        Prefix to log lines before logged messages (default "> ")
  -numactors int
        Number of randomly generated actors. (default 10)
  -qps float
        QPS per thread. (default 100)
  -s int
        The size of save state value. (default 1024)
  -t duration
        How long to run the test. (default 1m0s)

...
```

The below examples shows the test results when `testclient` activates 4000 actors and generates load at 1000 max. qps using 5 threads.

```
$ dapr run --app-id testclient -- ./testclient -qps 1000 -c 5 -numactors 4000

== APP == 19:02:30 I main.go:77> activated actors: 4000
== APP == 19:02:30 I main.go:80> random payload: h145UAKXeOOY05SwlkAMCn62BukH2r7rKhsXDIjwNlFYWDwXCKejI3WOD9z6PDN3VqxldEFT1iRRxTgNd2ceA24oqA1ucxL80GZmeWm05oqHryQaQ50SIEDJ3B9xmKnUQ5EHCy2t8wCPizEy9xmr2RnR5ekJxXSxJhcOYGCkiLm05OxGcWmp3T7JXRRsrR2PgDLXm7WdTNRvNzzOdwSbfvWwN6v7tocCfilIhhOeolDpTgcC0w5u3aTHji2JxArf9QPmgtrPordHDbxThMM4D5WT2wY8y4FNil05MaZQ2WoU0ZHOiRp3S4Bt0pwuJzEiOaBk9CfSEhL5yRCvgVTzpEBuQuCo0nNcZyxabNtkxf0p2s8gKFFiIfrR1U72tYmTjbUGSPFrWvFBwFUY4qSl0kR0Zz7ZQUJwsl3F3WooGjklSrl2Cb7Jyr5AKjakivYS2MSXu2UBQNLHRhqoEhklQbvdpEFGYRZR9LIZiw3HKWQOObbxIZuSNtnR7QLLqd1J0FmXQhNhIrWhqPAxRkocuLKZ2zoIJcbktydLFp4v0WwuNm82eTOr2AdSC1uo1yJcOgWaCA7XrDWcTpNzWBUPXsb4fvSrkmSJnbQDX0rG9ujrkGIYkqa0n6FUFHNSFXIHo2t8PQB32uGSwNJOumiqAEt9qkqi27zDcszVQw2G6FfsGWxNpuosxVbFOxElHsqbypobvIfZP1RpHWuCLSvopXAO0P8Xh4Z5h9OLJIsuzRkL3qeZcj0zHvbWJSwdCvwLFIEgmfEwtkEFoOH2vdUeGOPzYqIeWjtHYr5y9eMsAIT088QcH9yg7pMHplUPJPPd9qpWwFEp34gct2YAwIyP8j6h5iajCavKdCLY9c6olj0lrF4ZkY5iyIJh1XrFjPhO8bjL1dLwFRrDGcCQpBU90f7qgmNffZVnPWxiSDnVeZfM0uqGoWSOngPJ0xT3EkZnbLsn5P6yLCLuDEwZF2yVeLo5ILS8dTFEEwod4Vck8ufssZTUjZjFWpwgSvE0vBNw
== APP == Starting at 1000 qps with 5 thread(s) [gomax 16] for 5s : 1000 calls each (total 5000)
== APP == 19:02:35 I periodic.go:558> T004 ended after 5.003374752s : 1000 calls. qps=199.86510097015415
== APP == 19:02:35 I periodic.go:558> T001 ended after 5.003405198s : 1000 calls. qps=199.86388477985508
== APP == 19:02:35 I periodic.go:558> T000 ended after 5.005255128s : 1000 calls. qps=199.79001557900207
== APP == 19:02:35 I periodic.go:558> T003 ended after 5.005303837s : 1000 calls. qps=199.78807132702744
== APP == 19:02:35 I periodic.go:558> T002 ended after 5.005312719s : 1000 calls. qps=199.78771680019779
== APP == Ended after 5.005343428s : 5000 calls. qps=998.93

== APP == Aggregated Sleep Time : count 4995 avg -0.0028920729 +/- 0.00482 min -0.03108044 max 0.002391638 sum -14.4459041
== APP == # range, mid point, percentile, count
== APP == >= -0.0310804 <= -0.001 , -0.0160402 , 53.49, 2672
== APP == > -0.001 <= 0 , -0.0005 , 70.25, 837
== APP == > 0 <= 0.001 , 0.0005 , 90.65, 1019
== APP == > 0.001 <= 0.002 , 0.0015 , 99.62, 448
== APP == > 0.002 <= 0.00239164 , 0.00219582 , 100.00, 19
== APP == # target 50% -0.0029652
== APP == WARNING 53.49% of sleep were falling behind
== APP == Aggregated Function Time : count 5000 avg 0.0047372677 +/- 0.001461 min 0.002284744 max 0.024743274 sum 23.6863383
== APP == # range, mid point, percentile, count
== APP == >= 0.00228474 <= 0.003 , 0.00264237 , 3.04, 152
== APP == > 0.003 <= 0.004 , 0.0035 , 31.52, 1424
== APP == > 0.004 <= 0.005 , 0.0045 , 67.70, 1809
== APP == > 0.005 <= 0.006 , 0.0055 , 87.46, 988
== APP == > 0.006 <= 0.007 , 0.0065 , 93.82, 318
== APP == > 0.007 <= 0.008 , 0.0075 , 96.96, 157
== APP == > 0.008 <= 0.009 , 0.0085 , 98.42, 73
== APP == > 0.009 <= 0.01 , 0.0095 , 99.30, 44
== APP == > 0.01 <= 0.011 , 0.0105 , 99.64, 17
== APP == > 0.011 <= 0.012 , 0.0115 , 99.72, 4
== APP == > 0.012 <= 0.014 , 0.013 , 99.88, 8
== APP == > 0.014 <= 0.016 , 0.015 , 99.90, 1
== APP == > 0.02 <= 0.0247433 , 0.0223716 , 100.00, 5
== APP == # target 90% 0.00639937
== APP == Jitter: false
== APP == Code 200 : 5000 (100.0 %)
== APP == Response Body/Total Sizes : count 5000 avg 5 +/- 0 min 5 max 5 sum 25000
== APP == 19:02:35 I main.go:176> Test is done
```

## Issue reference

#2002 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by youngbupark** [comment]
@yaron2 @amanbha while running this load test, I discovered the bug where this load test works only after I restarted placement service and dapr runtime for actor service in self-host mode. I can reproduce this in kubernetes. When I scale out client service, the first instance can call actor while the other instances are not able to invoke actor because they cannot find the actor.

Looks like @KaiWalter is facing the exactly same issue - https://github.com/dapr/dapr/issues/1732#issuecomment-691055389

It may be the regression of the fix #1851. I will look at it.

**Comment by artursouza** [comment]
I think the e2e and perf tests deserve a new repo at this point, not?

**Comment by youngbupark** [comment]
> I think the e2e and perf tests deserve a new repo at this point, not?

Personally, I would like to keep e2e and perf test in the same repo for now because it is easy to validate the current dapr/dapr change with this e2e tests... we can split it later. but I am not sure it is worth it now.

BTW, this PR includes only test application. I will add test driver code later to integrate with our e2e and perf test later.

**Comment by amanbha** [comment]
> > I think the e2e and perf tests deserve a new repo at this point, not?
> 
> Personally, I would like to keep e2e and perf test in the same repo for now because it is easy to validate the current dapr/dapr change with this e2e tests... we can split it later. but I am not sure it is worth it now.
> 
> BTW, this PR includes only test application. I will add test driver code later to integrate with our e2e and perf test later.

Yes, e2e and perf will run against the changes as they are happening in master branch, so we should keep them here atleast until 1.0

**Comment by artursouza** [comment]
You can accomplish the same even in different repos. We can revisit this later.

File: tests/apps/actorload/pkg/telemetry/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package telemetry
```
Comment (artursouza) [inline tests/apps/actorload/pkg/telemetry/telemetry.go:1]: License.

File: tests/apps/actorload/makefile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+export GO111MODULE ?= on
```
Comment (artursouza) [inline tests/apps/actorload/makefile:1]: nit: name file `Makefile`, with capital M.

File: tests/apps/actorload/cmd/stateactor/service/server.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
+package service
```
Comment (artursouza) [inline tests/apps/actorload/cmd/stateactor/service/server.go:1]: License.

File: tests/apps/actorload/cmd/stateactor/service/response.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package service
```
Comment (artursouza) [inline tests/apps/actorload/cmd/stateactor/service/response.go:1]: License.

**Review by artursouza — CHANGES_REQUESTED** [review]

**Review by amanbha — DISMISSED** [review]

File: tests/apps/actorload/pkg/telemetry/telemetry.go (hunk 1 above)
Comment (youngbupark) [reply tests/apps/actorload/pkg/telemetry/telemetry.go:6]: fixed.

File: tests/apps/actorload/makefile (hunk 2 above)
Comment (youngbupark) [reply tests/apps/actorload/makefile:1]: I cannot rename to Makefile :(

File: tests/apps/actorload/cmd/stateactor/service/server.go (hunk 3 above)
Comment (youngbupark) [reply tests/apps/actorload/cmd/stateactor/service/server.go:6]: added.

File: tests/apps/actorload/makefile (hunk 2 above)
Comment (youngbupark) [reply tests/apps/actorload/makefile]: ah I changed.

**Review by amanbha — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**